src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -195,6 +211,15 @@ private IHost BuildHost(bool isOffline = false)
                 });
             }
 
+            if (hasHostConfigErrors)
+            {
+                builder.ConfigureAppConfiguration(b =>
+                {
+                    var source = b.Sources.OfType<HostJsonFileConfigurationSource>().Single();
```

Comment:
Can we flow a configuration item that could be used by the provider instead of removing it from the collection?